<?xml version="1.0"?>
<!--Inspired from https://github.com/jhu-lcsr/aruco_hand_eye/blob/master/launch/aruco_hand_eye.launch-->
<launch>
  <arg name="interactive" default="false"/>
  <arg name="marker_size" default="0.144"/>
  <arg name="marker_id" default="582"/>
  <!-- Track the aruco target in a given camera -->
  <node name="aruco_tracker" pkg="aruco_ros" type="single">
    <!-- Information about the RGB Camera -->
    <!--<remap from="/camera_info" to="camera/rgb/camera_info"/>-->
    <remap from="/camera_info" to="kinect2/hd/camera_info"/>
    <!-- RGB Rectified Image topic  -->
    <!--<remap from="/image" to="camera/rgb/image_rect_color"/>-->
    <remap from="/image" to="kinect2/hd/image_color_rect"/>
    
    <param name="image_is_rectified" value="True"/>
    <param name="marker_size" value="$(arg marker_size)"/>
    <!-- in m -->
    <param name="marker_id" value="$(arg marker_id)"/>
    <!-- RGB Optical Frame -->
    <!--<param name="reference_frame" value="/camera_link"/>-->
    <param name="reference_frame" value="/kinect2_rgb_optical_frame"/>
    <!-- RGB Optical Frame (same as reference_frame) -->
    <!--<param name="camera_frame" value="/camera_rgb_optical_frame"/>-->
    <param name="camera_frame" value="/kinect2_rgb_optical_frame"/>
    <!-- Aruco Marker Reference Frame (position of the aruco marker in space) -->
    <param name="marker_frame" value="aruco_marker_frame"/>
  </node>
  <!-- Run the visp solver to determine the extrinsic parameters -->
  <node name="hand_eye_solver" pkg="visp_hand2eye_calibration" type="visp_hand2eye_calibration_calibrator">
    <remap from="/compute_effector_camera_quick" to="compute_effector_camera_quick"/>
  </node>
  <!-- Connect the aruco tracker to the solver -->
  <node name="hand_eye_connector" launch-prefix="" pkg="aruco_hand_eye" type="calibrate.py" output="screen">
    <param name="sample_rate" value="2"/>
    <!-- Reference Frame where the Camara is attached on the robot -->
    <!--<param name="camera_parent_frame" value="/tilt_assembly"/>-->
    <param name="camera_parent_frame" value="/ptu_mount_link"/>
    <!-- Reference Frame to where the Aruco Marker is going to be located (usually the base of the robot) -->
    <param name="marker_parent_frame" value="/base_link"/>
    <param name="publish_tf" value="true"/>
    <param name="tf_suffix" value=""/>
    <param name="interactive" value="$(arg interactive)"/>
    <rosparam param="xyz_optical_base" subst_value="true">[0.000, 0.0, 0.000]</rosparam>
    <rosparam param="rpy_optical_base" subst_value="true">[0.0, 0.000, 0.0]</rosparam>
    <!--<rosparam param="xyz_optical_base" subst_value="true">[0.000, 0.022, 0.000]</rosparam>-->
    <!--<rosparam param="rpy_optical_base" subst_value="true">[-1.571, -0.000, -1.571]</rosparam>-->
  </node>
</launch>
